### PR TITLE
Task - Implement Package Scanner #8

### LINKED ISSUE #8: Implement Package Scanner
Develop the PackageScanner to scan specified packages and find classes with defined annotations (e.g., @Component, @Bean) and create BeanDefinitions.

### PR DESCRIPTION
**Pull request by ShpakDmytro**

### CODE REVIEW

File: src/main/java/com/codeus/winter/config/WinterBeanDefinition.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+ * <p>
+ * This class is intended to describe metadata for a bean, including its class name,
+ * scope, dependencies, and other properties required for its lifecycle.
+ * </p>
+ */
+public class WinterBeanDefinition implements BeanDefinition {
+
+    private String beanClassName;
+    private String scope = SCOPE_SINGLETON; // Default to singleton
+    private boolean autowireCandidate = true;
```
Comment (khshanovskyi) [inline src/main/java/com/codeus/winter/config/WinterBeanDefinition.java]: i suppose if u use `@Inject` then it should be `injectCandidate`

File: src/main/java/com/codeus/winter/config/WinterBeanDefinition.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    private boolean primary = false;
+    private String factoryBeanName;
+    private String factoryMethodName;
+    private String initMethodName;
+    private String destroyMethodName;
+    private final List<String> dependsOn = new ArrayList<>();
+    /**
```
Comment (khshanovskyi) [inline src/main/java/com/codeus/winter/config/WinterBeanDefinition.java]: pls split in code in appropriate places with empty row

**Review by khshanovskyi — APPROVED** [review]

File: src/main/java/com/codeus/winter/config/WinterPackageBeanRegistration.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            WinterBeanDefinition beanDefinition = new WinterBeanDefinition();
+            beanDefinition.setBeanClassName(clazz.getName());
+            beanDefinition.setScope(BeanDefinition.SCOPE_SINGLETON);
+            beanDefinition.setAutowireCandidate(true);
+
+            if (!registry.containsBeanDefinition(beanName)) {
+                registry.registerBeanDefinition(beanName, beanDefinition);
+
+                try {
+                    Object beanInstance = clazz.getDeclaredConstructor().newInstance();
```
Comment (DimaGalchenko) [inline src/main/java/com/codeus/winter/config/WinterPackageBeanRegistration.java]: We don't need to create an instance of Bean. Only following steps should be implemented:
- Scan package
- Create BeanDefinitions
- Save BeanDefinitions to register

**Review by AndriyPavlyuk — APPROVED** [review]

File: src/main/java/com/codeus/winter/config/WinterBeanDefinition.java (hunk 2 above)
Comment (ShpakDmytro) [reply src/main/java/com/codeus/winter/config/WinterBeanDefinition.java]: Sure

Comment (ShpakDmytro) [reply src/main/java/com/codeus/winter/config/WinterBeanDefinition.java]: Of course

File: src/main/java/com/codeus/winter/config/WinterPackageBeanRegistration.java (hunk 3 above)
Comment (ShpakDmytro) [reply src/main/java/com/codeus/winter/config/WinterPackageBeanRegistration.java]: Agree

**Review by DimaGalchenko — APPROVED** [review]

**Pull request merged**